src/item.h

```diff
@@ -1163,6 +1163,10 @@ class item : public visitable
         * The ids of all the qualities this contains.
         */
         const std::map<quality_id, int> &quality_of() const;
+        /**
+        * True if item has given quality or charged_quality (regardless of current charge).
+        */
+        bool has_any_quality( const std::string &quality ) const;
```

Comment:
Putting this on item is possibly too much syntax sugar - it doesn't do anything with the item itself and if you have the item then you have direct access to it's type field in 6 characters, this overloads an already giant item class for not much gain considering you can just do `i.type->has_any_quality(...`

imo as a pure ui helper to do string matching this feels like it should just be a static function near the ui call, otoh there's a bunch of these hanging off public interfaces anyway

Reply:
Thanks for feedback, will do in the evening.

Reply:
I have done this and some simple similar cases.
I haven't done it for  | the reason
--- | ---
`base_damage_thrown` | it has TODO
count_by_charges | too many references
damage_min | consistency with `damage()` and `damage_max()`
damage_max | consistency with `damage()` and `damage_min()`